appengine/go11x/cloudsql/cloudsql.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Sample cloudsql demonstrates connection to a Cloud SQL instance from App Engine standard.
+package main
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+
+	// MySQL library, comment out for PostgreSQL
+	_ "github.com/go-sql-driver/mysql"
+	// PostgreSQL Library, uncomment to use.
+	// _ "github.com/lib/pq"
+)
+
+var db *sql.DB
+
+func main() {
+	db = GetDB()
+
+	http.HandleFunc("/", indexHandler)
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+
+	log.Printf("Listening on port %s", port)
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%s", port), nil))
+}
+
+// GetDB gets a connection to the database.
+func GetDB() *sql.DB {
+	var (
+		connectionName = mustGetenv("CLOUDSQL_CONNECTION_NAME")
+		user           = mustGetenv("CLOUDSQL_USER")
+		password       = os.Getenv("CLOUDSQL_PASSWORD") // NOTE: password may be empty
+	)
+
+	// MySQL Connection, comment out for PostgreSQL
+	// connection string format: USER:PASSWORD@unix(/cloudsql/)PROJECT_ID:REGION_ID:INSTANCE_ID/[DB_NAME]
+	dbURI := fmt.Sprintf("%s:%s@unix(/cloudsql/%s)/", user, password, connectionName)
+	conn, err := sql.Open("mysql", dbURI)
+
+	// PostgreSQL Connection, uncomment to use.
+	// connection string format: user=USER password=PASSWORD host=/cloudsql/PROJECT_ID:REGION_ID:INSTANCE_ID/[ dbname=DB_NAME]
+	// dbURI := fmt.Sprintf("user=%s password=%s host=/cloudsql/%s dbname=%s", user, password, connectionName)
+	// conn, err := sql.Open("postgres", dbURI)
+
+	if err != nil {
+		panic(fmt.Sprintf("GetDB: %v", err))
+	}
+
+	return conn
+}
+
+// indexHandler responds to requests with our list of available databases.
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.NotFound(w, r)
+		return
+	}
+
+	w.Header().Set("Content-Type", "text/plain")
+
+	rows, err := db.Query("SHOW DATABASES")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Could not query db: %v", err), 500)
+		return
+	}
+	defer rows.Close()
+
+	buf := bytes.NewBufferString("Databases:\n")
+	for rows.Next() {
+		var dbName string
+		if err := rows.Scan(&dbName); err != nil {
+			http.Error(w, fmt.Sprintf("Could not scan result: %v", err), 500)
```

Comment:
Same error comment.

Reply:
Rotated these to:

```go
log.Printf("Could not scan result: %v", err)
http.Error(w, "Internal Error", 500)
```

Of note this sample will be replaced by @kurtisvg as part of a broader overhaul.